utils/errors/errors.go

```diff
@@ -0,0 +1,41 @@
+package errors
+
+import (
+	"fmt"
+)
+
+// Error interface
+type Error interface {
```

Comment:
I would suggest that we do not follow this model.  The standard library already implements "Unwrap" which allows "cause" to be figured out.  Small example below:

```go
package main

import (
	"errors"
	"fmt"
)

type MyCustomError struct {
	cause error
	data  string
}

func NewMyCustomError(cause error, data string) *MyCustomError {
	return &MyCustomError{
		cause: cause,
		data:  data,
	}
}

func (mce *MyCustomError) Error() string {
	return mce.data
}

func (mce *MyCustomError) Unwrap() error {
	return mce.cause
}

func main() {

	deepErr := errors.New("first error")
	shallowErr := NewMyCustomError(deepErr, "second error")
	surfaceErr := fmt.Errorf("third error %w", shallowErr)

	var myErr *MyCustomError
	if errors.As(errors.Unwrap(surfaceErr), &myErr) {
		fmt.Println(myErr.Error())
	}
}
```
So long as we implement "Unwrap" on all of our custom errors (such as ErrorBundle below) we can use the standard library for keeping track of cause, and more importantly we can use As and Is on the errors for us to have better error type-checking in our error handling.

I feel that redefining the "Error" interface is going to be confusing, and we will not be able to easily use the stdlib functionality.